finagle-netty4/src/main/scala/com/twitter/finagle/netty4/channel/ServerBridge.scala

```diff
@@ -15,6 +17,17 @@ private[netty4] class ServerBridge[In, Out](
 
   def initChannel(ch: Channel): Unit = {
     val transport: Transport[In, Out] = transportFac(ch)
-    serveTransport(transport)
+    val sslHandler = ch.pipeline().get(classOf[SslHandler])
+    if(sslHandler != null ) {
+      sslHandler.handshakeFuture().addListener( new GenericFutureListener[NettyFuture[Channel]] {
+        override def operationComplete(future: NettyFuture[Channel]): Unit = {
+          if(future.isSuccess) {
```

Comment:
If I understand [netty code](https://github.com/netty/netty/blob/1419f5b601ec7f50938a5e745ec2312f881a580f/handler/src/main/java/io/netty/handler/ssl/SslHandler.java#L1397) correctly, it is not necessary for netty4. `SslHandler` will shutdown `SSLEngine` and context (and it will close the underlying channel)